### DISCUSSION TITLE
Get node coordinates?

### DISCUSSION THREAD

**Discussion by novaphys** [networkx/networkx#4763] [Q&A]
How do I get a list of node positions from randomly generated graphs? -  specifically their (x,y) coordinates  , without having to privously have specified said positions using the pos-keyword, and without rewriting the coordinates using something like spring_layout? I'm just looking for a list of coordinates of this perticular embedding that the computer generated: 

![example_graph](https://user-images.githubusercontent.com/83298285/116273770-58109180-a782-11eb-9165-87094fd64e64.JPG)

Hope you're all in good health

**Comment by dschult** [comment]
If you don't specify the `pos` argument, networkx runs `spring_layout` for you, but it doesn't save/return the results of `pos`.
So, if you thunk you might need them later, call `spring_layout` yourself, pass the result into the draw function and use it as needed.

**Reply by novaphys** [reply]
Yes, and maybe the random case was not the best example of what I'm trying to achieve sorry, I'll be a bit more specific; I have a graph that I genered (non randomly) so the placement of the nodes obeys a condition, so I don't know the specific position I only know it satisfies that condition, and if I get the coordinates by calling spring_layout it overwrites the positions that I created, can I get an array of the coordinates without overwriting them?

**Reply by dschult** [reply]
The constrained positions must be stored somewhere or `draw` won't use them. By that I mean, literally, if you don't specify a `pos` argument in your `draw` command then networkx will just use `spring_layout` to generate a `pos` array. Find where the coordinates are stored and use that to pass that info into the drawing function.

**Comment by croonos23** [comment]
Try with this: 

import networkx as nx
import matplotlib.pyplot as plt

G = nx.Graph()

G.add_edges_from([
("P","1"),
("P","2"),
("P","3"),
("P","4"),
("P","5"),
("P","6"),
("P","7")

])
pos = nx.planar_layout(G, scale=1000)
nx.draw_networkx_nodes(G,pos)
nx.draw_networkx_edges(G,pos,edgelist=G.edges(),edge_color="black")
nx.draw_networkx_labels(G,pos)
print(pos)
plt.show()


================================================

{'P': array([-1000.        ,  -333.33333333]), '1': array([ 523.80952381, -333.33333333]), '2': array([-238.0952381 ,  428.57142857]), '3': array([-111.11111111,  301.58730159]), '4': array([ 15.87301587, 174.6031746 ]), '5': array([142.85714286,  47.61904762]), '6': array([269.84126984, -79.36507937]), '7': array([ 396.82539683, -206.34920635])}
